Assets/BossRoom/Scripts/Shared/Net/NetworkVariableGUID.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.IO;
+using Unity.Netcode;
+using UnityEngine;
+
+namespace BossRoom
+{
+    [Serializable]
+    public class NetworkVariableGUID : NetworkVariableBase
```

Comment:
same here, why do we need a whole new network variable for this?

Reply:
@VALERE91Unity

Reply:
A GUID is managed so we cannot put it in a NetworkVariable<>

Reply:
There is an alternative way of implementing this. Createa struct `NetworkGUID` have it store the guid as 2 ulongs. That way it's an unmanaged struct. Have it implement INetworkSerializable and then put it into `NetworkVariable<T>` that's probably a bit cleaner than a custom netvar.

Reply:
I change it for the struct version :)